tests/e2e/component.go

```diff
@@ -474,4 +476,173 @@ func componentTests(componentCmdPrefix string) {
 			waitForDeleteCmd("odo project list", projName)
 		})
 	})
+
+	Context("Test odo push with --source and --config flags", func() {
+		//new clean project and context for each test
+		var project string
+		var context string
+
+		//  current directory and project (before eny test is run) so it can restored  after all testing is done
+		var originalDir string
+		var originalProject string
+
+		// Setup up state for each test spec
+		// create new project (not set as active) and new context directory for each test spec
+		// This is before every spec (It)
+		BeforeEach(func() {
+			project = helper.OcCreateRandProject()
+			context = helper.CreateNewContext()
+		})
+
+		// Clean up after the test
+		// This is run after every Spec (It)
+		AfterEach(func() {
+			helper.OcDeleteProject(project)
+			helper.DeleteDir(context)
+		})
+
+		Context("when project from KUBECONFIG is used", func() {
```

Comment:
this should be renamed now to something like "when --project is used"

Reply:
ack